Character.cpp

```diff
@@ -98,20 +99,42 @@ void Character::attackInternal(Character& other)
             b) your stats are boosted 10%
             c) the initial value of your stats is updated to reflect this boosted stat for the next time you defeat another character.
       */
-        assert(false);
+        // restore stats to initial value
+        if( hitPoints < *initialHitPoints )
+        {
+            hitPoints = *initialHitPoints;
+        }
+        if( armor < *initialArmorLevel )
+        {
+            armor = *initialArmorLevel;
+        }
+        if( attackDamage < *initialAttackDamage )
+        {
+            attackDamage = *initialAttackDamage;
+        }
+        
+        // boost by 10%
+        boostHitPoints(hitPoints*0.1);
+        boostArmor(armor*0.1) ;
+        boostAttackDamage(attackDamage*0.1);
+        
+        // update the intial stats
+        *initialHitPoints = hitPoints;
+        *initialArmorLevel = armor;
+        *initialAttackDamage = attackDamage;
```

Comment:
All of this is doing the same thing 3 times.  write a free function and call that free function 3 times.
- [ ] revise